src/Filesystem/Folder.php

```diff
@@ -194,16 +194,32 @@ public function read($sort = true, $exceptions = false, $fullPath = false)
             if ($fullPath) {
                 $name = $item->getPathname();
             }
+
             if ($item->isDir()) {
-                $dirs[] = $name;
+                $dirs[$item->getCTime()][] = $name;
             } else {
-                $files[] = $name;
+                $files[$item->getCTime()][] = $name;
```

Comment:
I understood what you wanted to tell me, thought you were talking about ::Time sorting not ::Name.
Lets see about these tests. They are still passed on my local win.